packages/bot-web-ui/src/components/self-exclusion/__tests__/self-exclusion.spec.tsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { render, screen, waitFor } from '@testing-library/react';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import SelfExclusion, { SelfExclusionForm } from '../self-exclusion';
+import { Formik } from 'formik';
+import userEvent from '@testing-library/user-event';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({}));
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Modal: ({ children, is_open }: { children: JSX.Element; is_open: boolean }) => is_open && <div>{children}</div>,
+}));
```

Comment:
@maryia-matskevich-deriv is it possible to achieve without mocking the modal?

Reply:
@shafin-deriv Unfortunately no, we should mock it, because without mocking the modal component we can see only such DOM tree:
```
    <body>
      <div />
    </body>
```
but we expect to render proper structure and test this branch case:
<img width="832" alt="Screenshot 2024-07-29 at 14 33 31" src="https://github.com/user-attachments/assets/34e9d45d-83e2-4f97-ab41-b6f4b1706b8d">

Reply:
@maryia-matskevich-deriv this is because Modal is using a portal and you need to have an element with id `modal_root` present in the DOM so that it can mount. You can check `load-modal.spec.tsx` or `save-modal.spec.tsx` for reference.

Reply:
We can mock here but if you see a function attached with toggleModal props and if we mock we lose the checking. I would suggest let's try without mocking if it's not working then leave it for now and we can check later.

Reply:
I've tried, thank you, good suggestion🙂